independent-projects/qute/core/src/main/java/io/quarkus/qute/Booleans.java

```diff
@@ -0,0 +1,56 @@
+package io.quarkus.qute;
+
+import io.quarkus.qute.Results.Result;
+import java.lang.reflect.Array;
+import java.math.BigDecimal;
+import java.util.Collection;
+import java.util.Map;
+
+public final class Booleans {
+
+    private static final Long LONG_ZERO = 0L;
+    private static final Integer INTEGER_ZERO = 0;
+    private static final Short SHORT_ZERO = 0;
+    private static final Byte BYTE_ZERO = 0;
+    private static final Double DOUBLE_ZERO = 0.0;
+    private static final Float FLOAT_ZERO = 0.0F;
+
+    private Booleans() {
+    }
+
+    /**
+     * A value is considered falsy if it's null, {@link Result#NOT_FOUND}, {code false}, an empty collection, an empty map, an
+     * empty array, an empty string/char sequence or a number equal to zero.
+     * 
+     * @param value
+     * @return {@code true} if the value is falsy
+     */
+    public static boolean isFalsy(Object value) {
+        if (value == null || Results.Result.NOT_FOUND.equals(value)) {
+            return true;
+        } else if (value instanceof Boolean) {
+            return !(Boolean) value;
+        } else if (value instanceof Collection) {
+            return ((Collection<?>) value).isEmpty();
+        } else if (value instanceof Map) {
+            return ((Map<?, ?>) value).isEmpty();
+        } else if (value.getClass().isArray()) {
+            return Array.getLength(value) == 0;
+        } else if (value instanceof CharSequence) {
+            return ((CharSequence) value).length() == 0;
+        } else if (value instanceof Number) {
+            return isZero((Number) value);
+        }
+        return false;
+    }
+
+    private static boolean isZero(Number number) {
+        if (number instanceof BigDecimal) {
```

Comment:
`BigInteger`?

Also perhaps `AtomicInteger` and `AtomicLong`?

Reply:
+1 for `BigInteger`. I'm not so sure about the atomic variants. I mean you can always test the current value...

Reply:
In fact, should this entire method be just

```java
private static boolean isZero(Number number) {
    return number.longValue() == 0L; // || number.doubleValue() == 0.0;
}
```

That would also cover numeric values from other libraries, such as Guava or Apache Commons Lang.

Reply:
Hm, `number.longValue() == 0L || number.doubleValue() == 0.0d` would not work. E.g. `0.2d` would be considered falsy (the first condition is true) which is not correct. Or am I missing something?

Reply:
@Ladicek What about this? -> force push

Reply:
It's a trap and I probably shouldn't have suggested this :-) Because `Atomic*` numbers are also `Number`s, so you'll have `Double` going through the `doubleValue()` path, while `AtomicDouble` going through the `longValue()` path :-)

In retrospective, this was probably a bad idea. (Too bad `Number` doesn't prescribe `isZero`!)

Reply:
Hm, what about using `new java.math.BigDecimal(number.toString())` for any other numbers?